src/ContentValidationListener.php

```diff
@@ -213,8 +213,20 @@ public function onRoute(MvcEvent $e)
         }
 
         $inputFilter->setData($data);
-        if ($inputFilter->isValid()) {
-            return;
+        /**
+         * A specific Filter may throw an Exception if an invalid value is given.
+         * Convert any Exception to return a failed validation response
+         */
+        try {
+            if ($inputFilter->isValid()) {
+                return;
+            }
+        } catch (FilterInvalidArgumentException $ex) {
+            return new ApiProblemResponse(
+                new ApiProblem(422, 'Invalid value supplied to filter', null, null, array(
+                    'validation_messages' => array($ex->getMessage()),
+                ))
+            );
```

Comment:
There are a few problems with this approach. First, it may not be an invalid value at all; it may be a misconfigured filter/validator. Second, passing the exception message into `validation_messages` could potentially leak environment information into the response, and should only be done when in development mode. Finally, as noted, if this isn't a validation issue, but a configuration issue, 422 is an incorrect status code, and the error should not be returned in the `validation_messages` array at all.